NitroxClient/Communication/Packets/Processors/CellEntitiesProcessor.cs

```diff
@@ -2,64 +2,42 @@
 using NitroxClient.GameLogic.Helper;
 using NitroxModel.DataStructures.Util;
 using NitroxModel.GameLogic;
-using NitroxModel.Logger;
+using NitroxModel.Helper;
 using NitroxModel.Packets;
-using System.Collections.Generic;
 using UnityEngine;
 
 namespace NitroxClient.Communication.Packets.Processors
 {
     class CellEntitiesProcessor : ClientPacketProcessor<CellEntities>
     {
-        private PacketSender packetSender;
-        private HashSet<string> alreadySpawnedGuids = new HashSet<string>();
-        
+        private readonly PacketSender packetSender;
+
         public CellEntitiesProcessor(PacketSender packetSender)
         {
             this.packetSender = packetSender;
         }
 
         public override void Process(CellEntities packet)
         {
-            foreach(Entity entity in packet.Entities)
+            foreach (Entity entity in packet.Entities)
             {
-                if(!alreadySpawnedGuids.Contains(entity.Guid))
+                Optional<GameObject> opGameObject = GuidHelper.GetObjectFrom(entity.Guid);
+                GameObject gameObject;
+
+                if (opGameObject.IsPresent())
                 {
-                    SpawnEntity(entity);
+                    gameObject = opGameObject.Get();
                 }
                 else
                 {
-                    UpdateEntityPosition(entity);
+                    Validate.IsTrue(entity.TechType != TechType.None);
```

Comment:
This isn't really exceptional (as of right now) and shouldn't block other entities from being processed.